docs/sentence_transformer/usage/mteb_evaluation.md

```diff
@@ -0,0 +1,136 @@
+
+This guide walks you through using MTEB **with SentenceTransformer models for post-training evaluation**. This is *not* designed for use during training loops. To fully integrate your model to `MTEB` you can follow this [guide](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_model.md)
+
+---
+
+## Installation
+
+Install MTEB and its dependencies:
+
+```bash
+pip install mteb
+```
+
+> This also installs `sentence-transformers`, `datasets`, and task-specific dependencies.
+
+---
+
+##  Quick Start: One-Line Evaluation
+
+```python
+from sentence_transformers import SentenceTransformer
+from mteb import MTEB
+
+model = SentenceTransformer("all-MiniLM-L6-v2")
+tasks = mteb.get_tasks(tasks=["STSBenchmark"])
+evaluation = MTEB(tasks=tasks)
+evaluation.run(model, output_folder="results/")
+```
+
+This evaluates your model on the **STS Benchmark**, a Semantic Textual Similarity dataset with human-annotated sentence pairs. Output is saved in `results/`.
+
+---
+
+##  Supported Task Types and Examples
+
+MTEB supports the following **task families**:
+
+### 1. **Semantic Textual Similarity (STS)**
+
+* **STSBenchmark**: Predict similarity score (0-5) for English sentence pairs.
+* **BIOSSES**: Biomedical sentence similarity.
+* **SICK-R**: Sentence Involving Compositional Knowledge with human ratings.
+
+### 2. **Classification**
+
+* **AmazonCounterfactualClassification**: Predict product sentiment in counterfactual reviews.
+* **TwitterSentiment**: Predict sentiment on tweets (positive/negative).
+* **MassiveIntentClassification**: Classify user intent in multi-lingual commands.
+
+### 3. **Retrieval (Information Retrieval)**
+
+* **TREC-COVID**: Retrieve COVID-19 papers for search queries.
+* **SciFact**: Retrieve abstracts relevant to scientific claims.
+* **NFCorpus**: Retrieve passages for health and nutrition questions.
+
+### 4. **Reranking**
+
+* **MSMARCO**: Rerank candidate passages by relevance to a query.
+* **StackExchangeReranking**: Rerank answers on StackExchange questions.
+
+### 5. **Bitext Mining**
+
+* **BUCC**: Identify aligned sentences in multilingual corpora.
+* **Tatoeba**: Match translations across 100+ languages.
+
+You can evaluate specific tasks or full categories:
+
+```python
+mteb.get_tasks(tasks=["SICK-R", "AmazonCounterfactualClassification"])
+mteb.get_tasks(task_types=["Classification", "Retrieval"])
+```
+
+---
+
+##  Customize Output and Results Handling
+
+To avoid writing results to disk:
+
+```python
+evaluation.run(model, output_folder=None)
+```
+
+To extract scores programmatically:
+
+```python
+results = evaluation.run(model, output_folder=None)
+
+from mteb import MTEBResults
+summary = MTEBResults(results).main_scores()
+print(summary)
```

Comment:
this kinda hides what happens - I would just extract the main score pr task and average them